tests/test_gls_fitter.py

```diff
@@ -60,6 +60,20 @@ def test_gls_fitter(self):
                     assert np.abs(v - val[0]) <= e, msg
                     assert np.abs(1 - val[1] / e) < 0.1, msg
 
+    def test_noise_design_matrix_index(self):
+        self.fit(full_cov=False)
+        ntmpar = len(self.f.model.free_params)
+        Mn = self.f.model.noise_model_designmatrix(self.f.toas)
+        M, params, units = self.f.get_designmatrix()
+        M_stack = np.hstack((M, Mn))
+        noise_dims = self.f.model.noise_model_dimensions(self.f.toas)
+        # This is how the gls fitter get the noise design matrix. I don't know
+        # if there is some other way to test the index.
```

Comment:
For noise model basis functions that are easy to describe - like the constant offset one, or ECORR for a simple data set (merge two fake data sets so the ECORR basis functions are pairs) - you can confirm that the basis functions are what you think they should be. I wrote some examples of this in the tests for the downhill fitters, if I recall correctly.

Reply:
The problem is to get the noise model design matrix values from the full design matrix (Timingmodel design matrix + noise model design matrix). The noise dim calculation is ok. The problem is at the fitter. I don't have a good way to get P0, and P1, listed below from the fitter.